### PR TITLE
Promote section in schema

### PR DESCRIPTION
**Pull request by nealedj**
See https://github.com/ONSdigital/eq-survey-runner/pull/1473

This alters the validator with the updated schema.

### CODE REVIEW

File: schemas/questionnaire_v1.json [hunk 1]
```diff
@@ -141,48 +113,71 @@
             "$ref": "common_definitions.json#/id"
           },
           "title": {
-            "type": "string"
-          },
-          "hide_in_navigation": {
-            "type": "boolean"
+            "type": "string",
+            "description": "Title for a given id."
```
Comment (jonnyshaw89) [inline schemas/questionnaire_v1.json]: This description isn't very descriptive.
Is this the title that will be displayed in the navigation?

File: tests/schemas/test_invalid_numeric_answers.json [hunk 2]
```diff
@@ -7,127 +7,130 @@
...
-                    "answers": [{
-                            "id": "answer-2",
-                            "label": "Invalid Range",
-                            "type": "Currency",
-                            "currency": "GBP",
-                            "max_value": {
-                                "answer_id": "answer-1",
-                                "exclusive": true
+    "sections": [{
+        "id": "section1",
```
Comment (jonnyshaw89) [inline tests/schemas/test_invalid_numeric_answers.json]: pattern for the rest of this schema is `section-1`

**Comment by jonnyshaw89** [comment]
Do not merge until https://github.com/ONSdigital/eq-survey-runner/pull/1473 is in.
Otherwise it will break all builds

File: app/validation/validator.py [hunk 3]
```diff
@@ -211,22 +211,29 @@ def validate_child_answers_define_parent(self, json_to_validate):
...
-    def _get_answers(self, survey_json):
-        for block in self._get_blocks(survey_json):
-            for question in block.get('questions', []):
-                for answer in question['answers']:
-                    yield answer
+    @classmethod
+    def _get_answers(cls, survey_json):
+        return itertools.chain.from_iterable(
+            question.get('answers', []) for question in cls._get_questions(survey_json)
+        )
```
Comment (shahi645) [inline app/validation/validator.py]: 👍

File: schemas/questionnaire_v1.json [hunk 4]
```diff
@@ -141,48 +113,71 @@
...
+            "type": "string",
+            "description": "Title used for the navigation container."
           },
-          "skip_conditions": {
-            "$ref": "common_definitions.json#/skip_conditions"
-          },
-          "routing_rules": {
-            "ref": "common_definitions.json#/routing_rules"
+          "title_from_answers": {
+            "type": "array",
```
Comment (shahi645) [inline schemas/questionnaire_v1.json]: When would this be used?

Comment (nealedj) [reply schemas/questionnaire_v1.json]: In repeating question groups. I.e. where a section is created for each household member the title would be made up of each member's first and last name.

Comment (shahi645) [reply schemas/questionnaire_v1.json]: Ahh yeah 🙄

**Comment by jonnyshaw89** [comment]
@shahi645 why is your face on these commits?

**Comment by shahi645** [comment]
@jonnyshaw89 i have no idea

**Comment by nealedj** [comment]
Somebody messed up the rebase and pushed it... fixed now :)

**Comment by jonnyshaw89** [comment]
@nealedj looks like your commit message is too long and being truncated.
You should shorten the title and move the extra detail to following lines.

**Pull request merged**